tests/README.md

```diff
@@ -0,0 +1,106 @@
+# Neogit Tests
+
+## Running Tests
+
+As a base requirement you must have `make` installed.
+
+Once `make` is installed you can run tests by entering `make test` in the top level directory of Neogit into your command line.
+
+## Adding a test dependency
+
+If you're adding a lua plugin dependency to Neogit and wish to test it, open `tests/init.lua` in your editor.
+
+Look for the following lines:
+
+```lua
+if os.getenv("CI") then
+  vim.opt.runtimepath:prepend(vim.fn.getcwd())
+  vim.cmd([[runtime! plugin/plenary.vim]])
+  vim.cmd([[runtime! plugin/neogit.lua]])
+else
+  ensure_installed("nvim-lua/plenary.nvim")
+  ensure_installed("nvim-telescope/telescope.nvim")
+end
+```
+
+As an example let's say we want to add [`vim-fugitive`](https://github.com/tpope/vim-fugitive) to our test dependencies. Our new dependency section would look like this:
+
+```lua
+if os.getenv("CI") then
+  vim.opt.runtimepath:prepend(vim.fn.getcwd())
+  vim.cmd([[runtime! plugin/plenary.vim]])
+  vim.cmd([[runtime! plugin/neogit.lua]])
+  vim.cmd([[runtime! plugin/fugitive.vim]])
+else
+  ensure_installed("nvim-lua/plenary.nvim")
+  ensure_installed("nvim-telescope/telescope.nvim")
+  ensure_installed("tpope/vim-fugitive")
+end
+```
+
+## Test Organization
+
+### Where do tests go?
+
+All tests are to be placed within the `tests/specs` directory, and placed mocking the path of the Neogit module the test is responsible for. For instance, say you wanted to test `lua/neogit/config.lua` then you would create the test file in `tests/specs/config_spec.lua` which mirrors the path in the main Neogit module.
+
+### Where do utility functions go?
+
+If you have any utility code that has to do with git, it should be placed in `tests/util/git_harness.lua`.
+
+If you have a generic utility function _only_ relevant for tests then it should go in `tests/util/util.lua`. If it is generic enough that it could be useful in the general Neogit code then a consideration should be made to place this utility code in `lua/neogit/lib/util.lua`.
+
+### Where should raw content files go?
+
+Raw content files that you want to test against should go into `tests/test_files`. If you have a raw file you'd like to use against in git, then you'll need to add it to the git repository within `tests/.repo`. This can be done by changing directory into `tests/.repo` and renaming `.git.orig` to `.git` then adding any relevant changes to that repository. Once you're done, make sure you rename `.git` back to `.git.org`.
```

Comment:
I would prefer `tests/fixtures/` or `tests/supports/ to `test/test_files/` - what do you think?

Reply:
Of the two I prefer 'fixtures'. Will refactor to that in a bit.

Reply:
@CKolkey Done.